packages/react/__tests__/custom-cache.js

```diff
@@ -65,21 +65,20 @@ test('Global should "inherit" sheet class from the cache', () => {
   )
 
   expect(safeQuerySelector('head')).toMatchInlineSnapshot(`
-    <head>
-      <style
-        data-emotion="test-global"
-        data-s=""
-      >
-        
-        /** test-global */body{width:0;}
-      </style>
-      <style
-        data-emotion="test"
-        data-s=""
-      >
-        
-        /** test */.test-1lrxbo5{color:hotpink;}
-      </style>
-    </head>
-  `)
+<head>
+  <style
+    data-href="1lrxbo5"
+    data-precedence="emotion-test"
+  >
+    .test-1lrxbo5{color:hotpink;}
+  </style>
+  <style
+    data-emotion="test-global"
+    data-s=""
+  >
+    
+    /** test-global */body{width:0;}
+  </style>
```

Comment:
their order flipped, we can fix this by using precedence with `-global` on global styles, that will allow people to control how they are grouped - they could render dummy styles in head with those precedence set and then React will insert new styles there

the problem is that with those hoistable styles... we lose the ability to unmount global styles, are we fine with that?

Reply:
> we lose the ability to unmount global styles, are we fine with that?

imo, yes, it's fine. I've never been a big fan of the unmounting styles is `Global`, it's kind of started out as "i guess we can do this so why not" but imo it's not really that valuable since people should just scope styles (whether that's literally with the `css` prop/etc. or with global styles but all the styles are scoped to some selector) and conditionally apply the selector rather than have conditional style insertion.